pkg/gateway/gateway.go

```diff
@@ -573,7 +573,14 @@ func (g *gatewayModule) SetNamedProxy(wlID string, config zos.GatewayNameProxy)
 	if cfg.Domain == "" {
 		return "", errors.New("node doesn't support name proxy (doesn't have a domain)")
 	}
-	if err := g.validateNameContract(config.Name, twinID); err != nil {
+
+	reservedName := config.Name
+	subdomains := strings.Split(config.Name, ".")
+	if len(subdomains) >= 2 {
+		reservedName = subdomains[len(subdomains)-1]
+	}
```

Comment:
there is no need for this if statement here this can be like
```suggestion
	reservedName = subdomains[len(subdomains)-1]
```